text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+Languages that have continued to require large numbers of third-party dependency sources for most programs have had [outsized supply chain security problems](#appendix-c) despite, in some cases, massive investment. There is no reason to believe that the situation with Rust will be any different if the ecosystem does not make a concerted effort to reduce the average number of third-party dependencies in its programs.
+
+The [Cargo Vet](https://mozilla.github.io/cargo-vet/) system was introduced to mitigate supply chain security issues in the Rust dependency ecosystem. While it's a helpful and laudable effort, it does not go far enough. Code audits vary in quality, even from programmers at large companies. They are not a substitute for trusting the entity that produced the code in the first place. Cargo Vet does not address questions about a project's security policies, processes, and resources. It also adds to the complexity of supply chain security for Rust programmers, which means that it's not widely used and contributes negatively to the ESL's secondary goal, improving the Rust development experience.
+
+The [blessed.rs website](https://blessed.rs/crates) is a "hand-curated guide to the crates.io ecosystem, helping you choose which crates to use." It allows Rust developers to look up suggested crates based on function descriptions, significantly aiding with the issue of discovery and improving the Rust development experience. It does not, in its current form, contribute significantly to addressing the security concerns described in this RFC, but could in theory add more substantial and regular vetting for trustworthiness.
+
+Implementing the already approved [namespaces for Rust crates RFC](https://rust-lang.github.io/rfcs/3243-packages-as-optional-namespaces.html) would be helpful in terms of both security and developer experience. It would contribute positively to addressing security concerns by allowing for simpler recognition of which crates have the same sources as well as providing the opportunity to programmatically restrict dependencies to a certain set of trusted sources. It would contribute to improving the Rust development experience by pushing the need for uniqueness in naming to the namespace, thus allowing for more functionally descriptive naming of crates.
+
+# Prior art
+[prior-art]: #prior-art
+
+## Similar Efforts
+
+Libraries similar to the vision for the ESL:
+
+* [Go Standard Library](https://pkg.go.dev/std)
+* [Swift Core Libraries](https://www.swift.org/documentation/core-libraries/)
+* [Objective-C Foundation](https://developer.apple.com/documentation/foundation/)
+* [C# .NET CLI Standard Libraries](https://en.wikipedia.org/wiki/Standard_Libraries_(CLI))
+* [Python Standard Library](https://docs.python.org/3/library/index.html)
```

Comment:
A comparison with Boost might be relevant as it kind of acts like this

Reply:
I don't see how any of these 5 are "similar effort" to the proposed RFC. These are just "battery-included" library which evolve together with the language. ESLs are updated independently (and more importantly for Rust, allows breaking changes "v2.0"). I think these are more relevant:

* C++ Boost
* Haskell platform (*which has been deprecated*)
* `golang.org/x/*`